sdk/python/feast/loaders/ingest.py

```diff
@@ -1,165 +1,39 @@
 GRPC_CONNECTION_TIMEOUT_APPLY = 300  # type: int
 FEAST_SERVING_URL_ENV_KEY = "FEAST_SERVING_URL"  # type: str
 FEAST_CORE_URL_ENV_KEY = "FEAST_CORE_URL"  # type: str
 BATCH_FEATURE_REQUEST_WAIT_TIME_SECONDS = 300
-KAFKA_CHUNK_PRODUCTION_TIMEOUT = 120  # type: int
+BATCH_INGESTION_PRODUCTION_TIMEOUT = 120  # type: int
 
 
-def _encode_pa_tables(
-    file: str, feature_set: str, fields: dict, ingestion_id: str, row_group_idx: int
-) -> List[bytes]:
+def _check_field_mappings(
+    column_names: List[str],
+    feature_table_name: str,
+    feature_table_field_mappings: Dict[str, str],
+) -> None:
     """
-    Helper function to encode a PyArrow table(s) read from parquet file(s) into
-    FeatureRows.
-
-    This function accepts a list of file directory pointing to many parquet
-    files. All parquet files must have the same schema.
-
-    Each parquet file will be read into as a table and encoded into FeatureRows
-    using a pool of max_workers workers.
-
-    Args:
-        file (str):
-            File directory of all the parquet file to encode.
-            Parquet file must have more than one row group.
-
-        feature_set (str):
-            Feature set reference in the format f"{project}/{name}".
-
-        fields (dict[str, enum.Enum.ValueType]):
-            A mapping of field names to their value types.
-
-        ingestion_id (str):
-            UUID unique to this ingestion job.
+        Checks that all specified field mappings in FeatureTable can be found in
+        column names of specified ingestion source.
 
-        row_group_idx(int):
-            Row group index to read and encode into byte like FeatureRow
-            protobuf objects.
-
-    Returns:
-        List[bytes]:
-            List of byte encoded FeatureRows from the parquet file.
+        Args:
+            column_names: Column names in provided ingestion source
+            feature_table_name: Name of FeatureTable
+            feature_table_field_mappings: Field mappings of FeatureTable
     """
-    pq_file = pq.ParquetFile(file)
-    # Read parquet file as a PyArrow table
-    table = pq_file.read_row_group(row_group_idx)
-
-    # Add datetime column
-    datetime_col = pa_column_to_timestamp_proto_column(table.column(DATETIME_COLUMN))
-
-    # Preprocess the columns by converting all its values to Proto values
-    proto_columns = {
-        field_name: pa_column_to_proto_column(dtype, table.column(field_name))
-        for field_name, dtype in fields.items()
-    }
-
-    # List to store result
-    feature_rows: List[bytes] = []
 
-    # Loop optimization declaration(s)
-    field = FieldProto.Field
-    proto_items = proto_columns.items()
-    append = feature_rows.append
-
-    # Iterate through the rows
-    for row_idx in range(table.num_rows):
-        feature_row = FeatureRow(
-            event_timestamp=datetime_col[row_idx],
-            feature_set=feature_set,
-            ingestion_id=ingestion_id,
+    if "datetime" not in column_names:
```

Comment:
Should this be hardcoded? Dont they set the timestamp column themselves?

Reply:
You mean the user themselves should set the timestamp column?

Reply:
I guess you can use `timestampColumn` from FeatureTable spec

Reply:
The only thing we may wanna add `date_partition_column` ourselves here as
`df[<date_partition_column>] = to_date(df[<timestampColumn>])`
if it's specified and then use this column for partitioning

Reply:
You mean initialize `df[<date_partition_column>]` column if `date_partition_column` is specified in the FeatureTableSpec but not found in the dataframe?